src/protocolgame.cpp

```diff
@@ -1972,6 +2038,161 @@ void ProtocolGame::sendMarketDetail(uint16_t itemId)
 	writeToOutputBuffer(msg);
 }
 
+void ProtocolGame::sendStoreError(StoreError_t errorType, const std::string& message) {
+	NetworkMessage msg;
+	msg.addByte(0xE0);
+	msg.addByte(errorType);
+	msg.addString(message);
+
+	writeToOutputBuffer(msg);
+}
+
+void ProtocolGame::requestPurchaseData(uint32_t offerId, StoreOfferType_t offerType) {
+	NetworkMessage msg;
+	msg.addByte(0xE1);
+	msg.add<uint32_t>(offerId);
+	msg.addByte(offerType);
+
+	writeToOutputBuffer(msg);
+}
+
+void ProtocolGame::sendCoinBalance() {
+	NetworkMessage msg;
+	msg.addByte(0xF2);
```

Comment:
Seeing to the amount of bytes here, you should write down what they are.
Also you should make a new line for each new header that you send.